rs/pocket_ic_server/src/pocket_ic.rs

```diff
@@ -1699,6 +1706,206 @@ impl PocketIcSubnets {
         }
     }
 
+    fn deploy_ii(&self) {
+        // Nothing to do if the II subnet does not exist (yet).
+        let Some(ref ii_subnet) = self.ii_subnet else {
+            return;
+        };
+
+        if !ii_subnet
+            .state_machine
+            .canister_exists(IDENTITY_CANISTER_ID)
+        {
+            // Create the cycles ledger with its ICP mainnet settings.
+            // These settings have been obtained by calling
+            // `dfx canister call r7inp-6aaaa-aaaaa-aaabq-cai canister_status '(record {canister_id=principal"rdmx6-jaaaa-aaaaa-aaadq-cai";})' --ic`:
+            //     settings = record {
+            //       freezing_threshold = opt (2_592_000 : nat);
+            //       wasm_memory_threshold = opt (0 : nat);
+            //       controllers = vec { principal "r7inp-6aaaa-aaaaa-aaabq-cai" };
+            //       reserved_cycles_limit = opt (5_000_000_000_000 : nat);
+            //       log_visibility = opt variant { controllers };
+            //       wasm_memory_limit = opt (3_221_225_472 : nat);
+            //       memory_allocation = opt (0 : nat);
+            //       compute_allocation = opt (0 : nat);
+            //     };
+            let settings = CanisterSettingsArgs {
+                controllers: Some(BoundedVec::new(vec![ROOT_CANISTER_ID.get()])),
+                compute_allocation: Some(0_u64.into()),
+                memory_allocation: Some(0_u64.into()),
+                freezing_threshold: Some(2_592_000_u64.into()),
+                reserved_cycles_limit: Some(5_000_000_000_000_u128.into()),
+                log_visibility: Some(LogVisibilityV2::Controllers),
+                wasm_memory_limit: Some(3_221_225_472_u64.into()),
+                wasm_memory_threshold: Some(0_u64.into()),
+                environment_variables: None,
+            };
+            // Create the II canister.
+            // Although the II canister is created on a system subnet, it needs cycles:
+            // - the canister should have enough cycles to never run out of cycles;
+            // - it should still be possible top up the canister with further cycles without overflowing 128-bit integer range.
+            // The initial amount of ICP cycles equal to `u128::MAX / 2` satisfies both requirements.
+            let canister_id = ii_subnet.state_machine.create_canister_with_cycles(
+                Some(IDENTITY_CANISTER_ID.get()),
+                Cycles::from(u128::MAX / 2),
```

Comment:
Why does the II canister need cycles? (I saw an error log when the II canister was created with no cycles.)

Reply:
What was the error about? Is storing the config into state cost cycles? There is also an initialization of a salt, could that be it?

Reply:
Ah, with Google it tries to fetch Google certificates. Could that be it?

Reply:
Is the II canister making a canister http outcall? Canister http outcalls are free on a system subnet, but I guess you're using the Rust CDK (the `ic-cdk` crate) that does not distinguish between system and application subnets.

Now in that case (the II canister making canister http outcalls) the flow would only work in the so-called "live" mode of PocketIC in which canister http outcalls to remote servers are handled by PocketIC. So this brings me to the question if the II canister would work if `openid_google` is set to `None` (this is what I'd have to do in case the PocketIC instance is not in the "live" mode and canister http outcalls are not handled by PocketIC) - is that the case?